jax/_src/scipy/special.py

```diff
@@ -89,7 +89,7 @@ def erfinv(x):
 @api.custom_jvp
 @_wraps(osp_special.logit, update_doc=False)
 def logit(x):
-  x = asarray(x)
```

Comment:
What should really happen here is to call `_check_arraylike` on the input, so that list inputs will lead to an error. The attached bug is one of the reasons we started enforcing that.

Reply:
See #7737 for details

Reply:
IIUC, you want to add a `_check_arraylike` at the beginning of function?
But other functions in this file doesn't have this check now.
Maybe we can add it for all functions in this file in another PR?

Reply:
Why not this PR? If I understand the other issue correctly, it's that passing a list to the gradient of the function returns a confusing error – the reason that error is confusing is because we haven't called `_check_arraylike` at the start of the function.

Reply:
Making it accept a list is not the long-term behavior we want, so I think we should avoid a change that allows that input.

Reply:
The reason we haven't done this wholesale thorugh the package yet is that (1) we don't have any mechanism to automatically detect where the check is missing, and (2) each change introduces a backward-incompatibility for some small percentage of users, so it often requires some mitigation to land the change.

Reply:
I think the proposed implementation will reject (though not clearly) a list input. Since `lax_internal._convert_element_type` in `_promote_args_inexact` won't accept a list.
Almost all functions in this file use this promotion, maybe we can add the check for them all in this PR without breaking backward-compatibility?
Could you please check above claim?

Reply:
Ah, yes you're right, `_promote_args_inexact` does call `_check_arraylike`, so this change will work. Thanks!